ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineNewPayloadV3Test.java

```diff
@@ -0,0 +1,164 @@
+  }
+
+  @Test
+  public void beforeCancun_0x00DataFields_NilVersionedHashes_invalid() {
+    DataGas excessDataGas = DataGas.ZERO;
+    Long dataGasUsed = 0L;
+    List<String> versionedHashes = List.of();
+    var response = responsePayload(excessDataGas, dataGasUsed, versionedHashes);
+    Assertions.assertThat(response).isInstanceOf(JsonRpcErrorResponse.class);
+    final JsonRpcErrorResponse errorResponse = (JsonRpcErrorResponse) response;
+    assertThat(errorResponse.getError()).isEqualTo(JsonRpcError.UNSUPPORTED_FORK);
+  }
+
+  @Test
+  public void beforeCancun_NilExcessDataGas_0x00DataGasUsed_EmptyVersionedHashes_invalid() {
+    DataGas excessDataGas = DataGas.ZERO;
+    Long dataGasUsed = null;
+    List<String> versionedHashes = List.of();
+    var response = responsePayload(excessDataGas, dataGasUsed, versionedHashes);
+    Assertions.assertThat(response).isInstanceOf(JsonRpcErrorResponse.class);
+    final JsonRpcErrorResponse errorResponse = (JsonRpcErrorResponse) response;
+    assertThat(errorResponse.getError()).isEqualTo(JsonRpcError.UNSUPPORTED_FORK);
+  }
+
+  @Test
+  public void afterCancun_NilExcessDataGas_0x00DataGasUsed_EmptyVersionedHashes_invalid() {
+    mockAfterCancun();
+    DataGas excessDataGas = null;
+    Long dataGasUsed = 0L;
+    List<String> versionedHashes = List.of();
+    var response = responsePayload(excessDataGas, dataGasUsed, versionedHashes);
+    Assertions.assertThat(response).isInstanceOf(JsonRpcErrorResponse.class);
+    final JsonRpcErrorResponse errorResponse = (JsonRpcErrorResponse) response;
+    assertThat(errorResponse.getError()).isEqualTo(JsonRpcError.INVALID_PARAMS);
+  }
+
+  @Test
+  public void afterCancun_0x00ExcessDataGas_NilDataGasUsed_EmptyVersionedHashes_invalid() {
+    mockAfterCancun();
+    DataGas excessDataGas = DataGas.ZERO;
+    Long dataGasUsed = null;
+    List<String> versionedHashes = List.of();
+    var response = responsePayload(excessDataGas, dataGasUsed, versionedHashes);
+    Assertions.assertThat(response).isInstanceOf(JsonRpcErrorResponse.class);
+    final JsonRpcErrorResponse errorResponse = (JsonRpcErrorResponse) response;
+    assertThat(errorResponse.getError()).isEqualTo(JsonRpcError.INVALID_PARAMS);
+  }
+
+  JsonRpcResponse responsePayload(
+      final DataGas excessDataGas, final Long dataGasUsed, final List<String> versionedHashes) {
+    BlockHeader mockHeader =
+        new BlockHeaderTestFixture()
+            .excessDataGas(excessDataGas)
+            .dataGasUsed(dataGasUsed)
+            .buildHeader();
+
+    EnginePayloadParameter payload = mockPayload(mockHeader, List.of());
+    return method.response(
+        new JsonRpcRequestContext(
+            new JsonRpcRequest(
+                "2.0",
+                RpcMethod.ENGINE_NEW_PAYLOAD_V3.getMethodName(),
+                new Object[] {payload, versionedHashes})));
+  }
+
+  @Override
+  @Ignore
+  public void shouldRespondWithSyncingDuringBackwardsSync() {}
+}
```

Comment:
Is this hiding a bug?

Reply:
This has been moved up to abstract parent and is no longer ignored.